membership_extension/models/membership_line.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import timedelta
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+from openerp.addons.membership.membership import STATE
+
+
+class MembershipLine(models.Model):
+    _inherit = "membership.membership_line"
+    _order = 'date_to desc, id desc'
+
+    category_id = fields.Many2one(
+        string="Category", comodel_name='membership.membership_category',
+        related="membership_id.membership_category_id", readonly=True)
+    date_from = fields.Date(readonly=False)
+    date_to = fields.Date(readonly=False)
+    state = fields.Selection(
+        selection=STATE, readonly=False,
+        compute='_compute_state', inverse="_inverse_state",
+    )
+
+    def __init__(self, pool, cr):
```

Comment:
I tried with _register_hook, but it doesn't work because it is not called always between addons installations.
And I don't known why do you say that this is not multi-DB aware, because there is a pool per environment and one environment per database, so this __init__ is called always when this addon is loaded per each environtment.